### ISSUE TITLE
Error return nil

### ISSUE THREAD

**Issue by ripienaar** [antonmedv/expr#307]
This code worked in 1.9.0:

```golang
func TestExpr(t *testing.T) {
	env := map[string]any{
		"semver": func(value string, cmp string) (bool, error) { return true, nil },
	}

	prog, err := expr.Compile(`semver("1.2.3","= 1.2.3")`, expr.Env(env), expr.AsBool(), expr.AllowUndefinedVariables())
	if err != nil {
		t.Fatalf("compile failed: %v", err)
	}

	res, err := expr.Run(prog, env)
	if err != nil {
		t.Fatalf("could not execute compound query: %s", err)
	}

	if res != true {
		t.Fatalf("expected true got: %v", res)
	}
}
```

Should the `semver` func return error then the `expr.Run()` would fail with that error, great and when its nil the bool would be used - also great.  

Since 1.10.0 though when the error is nil the following happens:

```
compound_test.go:80: could not execute compound query: interface conversion: interface is nil, not error (1:1)
         | semver("1.2.3","= 1.2.3")
         | ^
```

So its not treating nil errors like nil

thanks for this amazing package!

**Comment by antonmedv** [comment]
I see, will try to understand the issues and find a solution later today.

**Comment by roqenrox** [comment]
The problem occurs in the following section:

```go
// https://github.com/antonmedv/expr/blob/9009d4da3de69a1e5dcbb284a9140129939de6f1/vm/vm.go#L303
out := fn.Call(in)
if len(out) == 2 && !runtime.IsNil(out[1]) {
	panic(out[1].Interface().(error))
}
```

On the other hand, it works as expected after adjusting the If condition as follows:

```go
// https://github.com/antonmedv/expr/blob/9009d4da3de69a1e5dcbb284a9140129939de6f1/optimizer/const_expr.go#L76
if len(out) == 2 && out[1].Type() == errorType && !out[1].IsNil() {
```

**Comment by antonmedv** [comment]
Fixed in b062494f3c863d0dfabf0eff469b42f995b0693e

Will create a new release.

**Comment by ripienaar** [comment]
I can confirm this is fixed, thanks a lot for this @antonmedv !
